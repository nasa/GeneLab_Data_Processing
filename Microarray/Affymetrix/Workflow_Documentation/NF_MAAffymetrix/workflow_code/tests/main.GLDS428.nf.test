nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "core"

    // test("GLDS-428: Should run without failures. First test case for Arabidposis Thaliana Biomart support") {

    //     when {
    //         params {
    //             // define parameters here. Example: 
    //             osdAccession = "OSD-428"
    //             gldsAccession = "GLDS-428"
    //             biomart_attribute = "AFFY AraGene"
    //         }
    //     }

    //     then {
    //         assert workflow.success
    //         assert snapshot(
    //                         path("${ params.outputDir }/Metadata").list(),
    //                         path("${ params.outputDir }/00-RawData").list(),
    //                         path("${ params.outputDir }/01-oligo_NormExp").list(),
    //                         path("${ params.outputDir }/02-limma_DGE").list(),
    //                         ).match()
    //     }

    // }

    test("GLDS-428: Local ISA Archive Version: Should run without failures.") {

        when {
            params {
                // define parameters here. Example: 
                osdAccession = "OSD-428"
                gldsAccession = "GLDS-428"
                isaArchivePath = "${ baseDir }/tests/assets/CLEANED_GLDS-428_metadata_GLDS-428-ISA.zip"
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                            path("${ params.outputDir }/Metadata").list(),
                            path("${ params.outputDir }/00-RawData").list(),
                            path("${ params.outputDir }/01-oligo_NormExp").list(),
                            path("${ params.outputDir }/02-limma_DGE").list(),
                            [ 'Affymetrix.html exists' : path("${ params.outputDir }/GeneLab/Affymetrix.html").exists()],
                            ).match()
        }

    }


}
