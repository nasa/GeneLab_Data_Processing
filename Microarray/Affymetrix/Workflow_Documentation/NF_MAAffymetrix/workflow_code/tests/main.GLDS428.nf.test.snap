{
    "GLDS-428: Should run without failures. First test case for Arabidposis Thaliana Biomart support": {
        "content": [
            [
                "GLDS-428_metadata_GLDS-428-ISA.zip:md5,fdaed5f7172d0c67b35da3d8194eb0ef",
                "OSD-428_microarray_v0_runsheet.csv:md5,72a1b5f955b3dea320ca43a47a21914a"
            ],
            [
                "raw_intensities_probe.csv:md5,aa475c65ef4a4f3a5273db890518bff2"
            ],
            [
                "normalized_expression_probeset.csv:md5,521f8dee0e92017863d37c2ceebfe090",
                "normalized_intensities_probe.csv:md5,410dcdcbc2d3ac24d1827b2dd085e0c7"
            ],
            [
                "SampleTable.csv:md5,1f6c9b6c611b7eb05f93526a0e22c24d",
                "contrasts.csv:md5,b80c6dd982176be22fd93110c78cc747",
                "differential_expression.csv:md5,211a0c972d2e2e58b867800647d2cfc7",
                "visualization_PCA_table.csv:md5,97804c3f739651a4849483fddf3ee8f2"
            ]
        ],
        "timestamp": "2023-05-17T19:00:02+0000"
    },
    "GLDS-428: Local ISA Archive Version: Should run without failures.": {
        "content": [
            [
                "OSD-428_microarray_v0_runsheet.csv:md5,bea8d40c0e870b363b550dafc46c9223"
            ],
            [
                "raw_intensities_probe.csv:md5,aa475c65ef4a4f3a5273db890518bff2"
            ],
            [
                "normalized_expression_probeset.csv:md5,521f8dee0e92017863d37c2ceebfe090",
                "normalized_intensities_probe.csv:md5,410dcdcbc2d3ac24d1827b2dd085e0c7"
            ],
            [
                "SampleTable.csv:md5,1f6c9b6c611b7eb05f93526a0e22c24d",
                "contrasts.csv:md5,b80c6dd982176be22fd93110c78cc747",
                "differential_expression.csv:md5,211a0c972d2e2e58b867800647d2cfc7",
                "visualization_PCA_table.csv:md5,97804c3f739651a4849483fddf3ee8f2"
            ],
            {
                "Affymetrix.html exists": false
            }
        ],
        "timestamp": "2023-05-17T19:00:02+0000"
    }
}